src/boostmovehub.ts

```diff
@@ -34,19 +34,15 @@ export class BoostMoveHub extends LPF2Hub {
     protected _currentPort = 0x3b;
     protected _voltagePort = 0x3c;
 
-    constructor (device: IBLEDevice, autoSubscribe: boolean = true) {
-        super(device, autoSubscribe);
-        this.type = Consts.HubType.BOOST_MOVE_HUB;
-        this._ports = {
-            "A": new Port("A", 0),
-            "B": new Port("B", 1),
-            "C": new Port("C", 2),
-            "D": new Port("D", 3),
-            "TILT": new Port("TILT", 58)
+    constructor (device: IBLEAbstraction) {
+        super(device, Consts.HubType.BOOST_MOVE_HUB);
+        this._portNames = {
+            "A": 0,
+            "B": 1,
+            "C": 2,
+            "D": 3,
+            "TILT": 58
```

Comment:
I think port names/id should appear in consts as they are constant and the same over same type hubs. It would also help developers like @ractive and me determine which port we should look at in our projects.

Reply:
I've now created a const (exported under the hub namespace) for each hub, accessible via `PoweredUP.WeDo2SmartHub.PortMap`, etc. This is also passed to the root hub object at constructor time. It's also available at runtime via a `ports` getter, ie. `hub.ports = ["A", "B", ...]`.